### PR TITLE
CHANGE: Add automated testing setup for InputForUI

### PR DESCRIPTION
**Pull request by jfreire-unity**
### Description

Adds the test setup for the InputSystemProvider that is used on the 2023.2+ InputForUI module.

### Changes made

Sets up the Unity project to test the InputForUI module with the InputSystemProvider implementation.
Adds a basic test with a mouse to validate that the test setup is working and that events are being dispatched.

### Notes

**This depends on #1679 and should not be merged before** 

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [x] Tests added/changed, if applicable.
    - Functional tests : 'Plugins/InputForUITests.cs`

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsTreeViewItem.cs
Comment (lyndon-unity) [inline Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsTreeViewItem.cs:64]: Great to see this fix - thank you :)

**Review by lyndon-unity — APPROVED** [review]
Looks like a sensible addition.
Obviously wait for the main provider to land (or merge into that branch)

**Comment by lyndon-unity** [comment]
That said this is failing tests which are related so it looks like it needs a fix

**Review by lyndon-unity — CHANGES_REQUESTED** [review]
Tests are failing ATM

error CS0122: 'EventProvider' is inaccessible due to its protection level

**Comment by jfreire-unity** [comment]
@lyndon-unity Yeah, this PR requires some `trunk` changes to land first.
Currently it's also a Draft PR. Once changes land in `trunk`, I'll re-run the tests and ask again for review.

File: Assets/Tests/InputSystem/Plugins/InputForUITests.cs
Comment (lyndon-unity) [inline Assets/Tests/InputSystem/Plugins/InputForUITests.cs:83]: Should we be testing a mouse click too here ?

**Review by lyndon-unity — APPROVED** [review]
Seems reasonable, happy to see test coverage expanding

**Comment by jfreire-unity** [comment]
Changing it back to DRAFT so that it only lands in 1.6.1

**Comment by jfreire-unity** [comment]
Holding this for after the 1.6.0 release since the `trunk` PR has not landed yet.

**Pull request merged**
